lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,191 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+Set-StrictMode -Version 2
+
+function Compare-XmlDocs($actual, $expected) {
+    if ($actual.Name -ne $expected.Name) {
+        throw "Actual name not same as expected: actual=" + $actual.Name + ", expected=" + $expected.Name
+    }
+    ##attributes...
+
+    if ($actual.Attributes -ne $null -and $expected.Attributes -ne $null) {
+        if ($actual.Attributes.Count -ne $expected.Attributes.Count) {
+           throw "attribute mismatch for actual=" + $actual.Name
+        }
+        for ($i=0;$i -lt $expected.Attributes.Count; $i =$i+1) {
+            if ($expected.Attributes[$i].Name -ne $actual.Attributes[$i].Name) {
+                throw "attribute name mismatch for actual=" + $actual.Name
+            }
+            if ($expected.Attributes[$i].Value -ne $actual.Attributes[$i].Value) {
+                throw "attribute value mismatch for actual=" + $actual.Name
+            }
+        }
+    }
+
+    ##children
+    if ($expected.ChildNodes.Count -ne $actual.ChildNodes.Count)  {
+        throw "child node mismatch. for actual=" + $actual.Name
+    }
+
+    for ($i=0;$i -lt $expected.ChildNodes.Count; $i =$i+1) {
+        if (-not $actual.ChildNodes[$i]) {
+            throw "actual missing child nodes. for actual=" + $actual.Name
+        }
+        Compare-XmlDocs $expected.ChildNodes[$i] $actual.ChildNodes[$i]
+    }
+
+    if ($expected.InnerText) {
+        if ($expected.InnerText -ne $actual.InnerText) {
+           throw "inner text mismatch for actual=" + $actual.Name
+        }
+    }
+    elseif ($actual.InnerText) {
+        throw "actual has inner text but expected does not for actual=" + $actual.Name
+    }
+}
+
+function FindElementsByName($node, $name) {
+    $elements = @()
+    foreach ($n in $node.ChildNodes) {
+        if ($n.Name -eq $name) {
+            $elements += $n
+        }
+    }
+    return ,$elements
+}
+
+function BackupFile($path) {
+	$backuppath = $path + "." + [DateTime]::Now.ToString("yyyyMMdd-HHmmss");
+	Copy-Item $path $backuppath;
+	return $backuppath;
+}
+
+$params = Parse-Args $args -supports_check_mode $true
+$dest = Get-AnsibleParam $params "path" -type "path" -FailIfEmpty $true -aliases "dest", "file"
+$fragment = Get-AnsibleParam $params "fragment" -type "str" -FailIfEmpty $true -aliases "xmlstring"
+$root = Get-AnsibleParam $params "root" -type "str" -FailIfEmpty $false -Default "DocumentElement" -aliases "xpath"
+$backup = Get-AnsibleParam $params "backup" -type "bool" -FailIfEmpty $false -Default "no" -ValidateSet "no", "yes"
+$type = Get-AnsibleParam $params "type" -type "str" -FailIfEmpty $false -Default "element" -ValidateSet "element", "attribute", "text"
+$attribute = Get-AnsibleParam $params "attribute" -type "str" -FailIfEmpty ($type -eq "attribute")
+
+$result = New-Object PSObject @{
+    win_xml = New-Object PSObject
```

Comment:
Please don't add the module-name as part of the result. This makes everything harder for users to use.
We put all the result data directly as part of the result hash table.